CHANGELOG.md

```diff
@@ -1,3 +1,54 @@
+## `5.0.0` Welp Welp Welp
+So it turns out Choo could be radically simplified. We're now comfortably
+sitting at `~4kb`, have removed a whole bunch of words from the API and should
+be a whole lot faster. We've [written about it
+before](https://medium.com/@yoshuawuyts/choo-v5-bc775b007b5e); if you're
+interested we recommend reading through that post.
+
+We're now using an event emitter, mutable state and explicit re-renders. Some
+people might frown at first at the words "mutable state", but if you think it
+through the mental model doesn't change. "State" has always been a concept of
+an object that changes over time; we then render the DOM as a snapshot of that
+state.
+
+What we've done is change the way we mutate that state - we no longer generate
+a ton of expensive intermediate objects to mutate the state, but instead mutate
+the state directly. In turn we've also removed the magic re-rendering and made
+it explicit. This enables people to create tight render loops that can even be
+used in GC constrained environments like games or music production. We think
+this change was well worth it, and will make a lot of sense going forward.
+
+People might also wonder why we've moved away from `flux`/`elm` and are using
+an event emitter now. It turns out that the previous architecture had a lot of
+confusing words that made it harder to learn than it should. It was also not
+possible to react to changes; the thing that changed always had to specify what
+needed to respond to it. By using event emitters we've inverted this, which
+will make relations in the application more expressive. All in all, it turned
+out that all we needed for this was a simple event emitter - we think this was
+well worth the change and breaking away from what we were previously doing.
+
+_Pretty much everything about the API changed in this version. There's
+literally nothing left to remove from the API tho so this is probably the last
+time we get to break anything in a significant way._
+
+### changes
+- :exclamation: state is now mutable and renders are triggered through
+  `.emit('render')`.
+- :exclamation: we've replaced `.use()`, `.model()` and the rest of the choo
+  architecture with a reworked `.use()` method. It's called once on boot, and
+  exposes a mutable reference to `state` and an event emitter that's compatible
+  with Node's `require('events').EventEmitter`
+- :exclamation: the `.router()` method has been replaced with `.route()`,
+  replacing the nested array API. This should be easier to remember and more
+  performant.
+- :exclamation: we've replaced `morphdom`/`yo-yo` with `nanomorph`. The two
+  algorithms are very comparable. The differences are that the new algorithm
+  is smaller and the value of input fields on re-rendering will be whatever the
+  `value=""` attribute is.
+- :exclamation: `choo/mount` is now available as `app.mount()`
```

Comment:
maybe worth noting that `app.mount` calls `start` internally, so we'll use one or the other, right?

compared to v4 where `mount` was called using the tree returned from `app.start`